routes/compat/wakatime/v1/leaders.go

```diff
@@ -0,0 +1,82 @@
+package v1
+
+import (
+	"github.com/go-chi/chi/v5"
+	"github.com/muety/wakapi/helpers"
+	"github.com/muety/wakapi/middlewares"
+	"github.com/muety/wakapi/models"
+	"github.com/muety/wakapi/utils"
+	"net/http"
+
+	conf "github.com/muety/wakapi/config"
+	v1 "github.com/muety/wakapi/models/compat/wakatime/v1"
+	"github.com/muety/wakapi/services"
+)
+
+type LeadersHandler struct {
+	config          *conf.Config
+	userSrvc        services.IUserService
+	leaderboardSrvc services.ILeaderboardService
+}
+
+func NewLeadersHandler(userService services.IUserService, leaderboardService services.ILeaderboardService) *LeadersHandler {
+	return &LeadersHandler{
+		userSrvc:        userService,
+		leaderboardSrvc: leaderboardService,
+		config:          conf.Get(),
+	}
+}
+
+func (h *LeadersHandler) RegisterRoutes(router chi.Router) {
+	router.Group(func(r chi.Router) {
+		r.Use(middlewares.NewAuthenticateMiddleware(h.userSrvc).Handler)
+		r.Get("/compat/wakatime/v1/leaders", h.Get)
+	})
+}
+
+// @Summary List of users ranked by coding activity in descending order.
+// @Description Mimics https://wakatime.com/developers#leaders
+// @ID get-wakatime-leaders
+// @Tags wakatime
+// @Produce json
+// @Security ApiKeyAuth
+// @Success 200 {object} v1.LeaderboardViewModel
+// @Router /compat/wakatime/v1/leaders [get]
+func (h *LeadersHandler) Get(w http.ResponseWriter, r *http.Request) {
+	wakapiUser, err := h.userSrvc.GetUserById("current")
```

Comment:
This will search for a user with "current" as its actual name, which won't work. Please check some of the other endpoints for an example of how to get the currently logged in user.

Reply:
Yep you're right :P. Thanks!